File: backend/src/main/java/com/webvrlamov/alstromeria/core/projection/model/ResponsePage.java
package com.webvrlamov.alstromeria.core.projection.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.domain.Sort;

import java.util.List;

@AllArgsConstructor
@Getter
@Setter
public class ResponsePage<E> {
    public long page;
    public long size;
    public long pagesCount;
    public long itemsCount;
    public Sort sort;
    public List<?> items;
}

File: frontend/src/app/service/http/model/response-page.ts
import {HasId} from "./pageable";

export interface ResponsePage<Item extends HasId> {
  page: number,
  size: number,
  pagesCount: number,
  itemsCount: number,
  sort: {
    unsorted: true,
    sorted: false,
    empty: true
  },
  items: Array<Item>
}
